templates/common/_base/files/etc-networkmanager-dispatcher.d-90-long-hostname.yaml

```diff
@@ -0,0 +1,40 @@
+filesystem: "root"
+mode: 0755
+path: "/etc/NetworkManager/dispatcher.d/90-long-hostname"
+contents:
+  inline: |
+    #!/bin/bash
+    #
+    # On Google Compute Platform (GCP) the hostname may be too long (>63 chars).
+    # During firstboot the hostname is set in the initramfs before NetworkManager
+    # runs.
+    #
+    # This hook is a simple work around: if the host name is longer than 63
+    # characters, then the hostname is truncated at the _first_ dot.
+    #
+    # Additionally, this hook does not break DNS or cluster resolution,
+    # since NetworkManager sets the appropriate /etc/resolv.conf settings.
+
+
+    IF=$1
+    STATUS=$2
+
+    log() { logger --tag "network-manager/$(basename $0)" "${@}"; }
+
+    if [ "${HOSTAME}" == "localhost" ]; then
+        log "hostname is already set"
+        exit 0
+    fi
+
+    if [[ ! "$STATUS" =~ (up|hostname|dhcp4-change|dhcp4-change) ]]; then
```

Comment:
dhcp4-change seems to be repeated.
Is this intentional ?

Reply:
Ah, thank you. That was supposed to be `dhcp6`. Changed and repushed.

Reply:
I don't see difference after refreshing several times.

Reply:
I still see `dhcp4-change` twice as well.  Forgot to commit or push?